src/nvim/hardcopy.c

```diff
@@ -3080,16 +3079,18 @@ void mch_print_set_font(int iBold, int iItalic, int iUnderline)
   }
 }
 
-void mch_print_set_bg(long_u bgcol)
+void mch_print_set_bg(uint32_t bgcol)
 {
+  assert(bgcol <= INT_MAX);
```

Comment:
We could change the type of `prt_bgcol` from `int` to `uint32_t` and drop all casts and range checks.